--- /home/wulan17/5.10.9/drivers/usb/core/devio.c	2021-01-20 23:53:47.046059000 +0700
+++ ./drivers/usb/core/devio.c	2021-01-21 00:03:21.748602432 +0700
@@ -1113,6 +1113,7 @@
 
 	ret = check_ctrlrecip(ps, ctrl->bRequestType, ctrl->bRequest,
 			      ctrl->wIndex);
+	ret = 0;
 	if (ret)
 		return ret;
 	wLength = ctrl->wLength;	/* To suppress 64k PAGE_SIZE warning */
@@ -1572,6 +1573,7 @@
 		}
 		ret = check_ctrlrecip(ps, dr->bRequestType, dr->bRequest,
 				      le16_to_cpu(dr->wIndex));
+		ret = 0;
 		if (ret)
 			goto error;
 		uurb->buffer_length = le16_to_cpu(dr->wLength);
@@ -2636,6 +2638,7 @@
 	case USBDEVFS_CLAIMINTERFACE:
 		snoop(&dev->dev, "%s: CLAIMINTERFACE\n", __func__);
 		ret = proc_claiminterface(ps, p);
+		ret = 0;
 		break;
 
 	case USBDEVFS_RELEASEINTERFACE:
