From 83ced08135dcccbfa17a5dfc471cd37603245bdb Mon Sep 17 00:00:00 2001
From: wulan17 <galihgustip@gmail.com>
Date: Tue, 2 Mar 2021 15:11:15 +0000
Subject: [PATCH] Allow Disable MSR Lockdown

---
 arch/x86/kernel/msr.c     |  2 ++
 security/lockdown/Kconfig | 12 ++++++++++++
 2 files changed, 14 insertions(+)

diff --git a/arch/x86/kernel/msr.c b/arch/x86/kernel/msr.c
index f7af978c8476..aa773279bf75 100644
--- a/arch/x86/kernel/msr.c
+++ b/arch/x86/kernel/msr.c
@@ -116,9 +116,11 @@ static ssize_t msr_write(struct file *file, const char __user *buf,
 	int err = 0;
 	ssize_t bytes = 0;
 
+#if defined(LOCK_DOWN_DENY_RAW_MSR)
 	err = security_locked_down(LOCKDOWN_MSR);
 	if (err)
 		return err;
+#endif
 
 	err = filter_write(reg);
 	if (err)
diff --git a/security/lockdown/Kconfig b/security/lockdown/Kconfig
index e84ddf484010..341f875d20d1 100644
--- a/security/lockdown/Kconfig
+++ b/security/lockdown/Kconfig
@@ -44,4 +44,16 @@ config LOCK_DOWN_KERNEL_FORCE_CONFIDENTIALITY
 	 code to read confidential material held inside the kernel are
 	 disabled.
 
+config LOCK_DOWN_DENY_RAW_MSR
+	bool "Lock down and deny raw MSR access"
+	depends on LOCK_DOWN_KERNEL_FORCE_CONFIDENTIALITY
+	default y
+	help
+	 Some Intel based systems require raw MSR access to use the flush
+	 MSR for MDS mitigation confirmation. Raw access can also be used
+	 to undervolt many Intel CPUs.
+
+	 Say Y to prevent access or N to allow raw MSR access for such
+	 cases.
+
 endchoice
-- 
2.25.1

