From c0dfadeba006e91dc97314e3c4b55ef1def401ab Mon Sep 17 00:00:00 2001
From: wulan17 <galihgustip@gmail.com>
Date: Tue, 2 Mar 2021 15:15:12 +0000
Subject: [PATCH] remove plus char from localversion

---
 scripts/setlocalversion | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index bb709eda96cd..1e49bc210f18 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -187,7 +187,7 @@ else
 	# LOCALVERSION= is not specified
 	if test "${LOCALVERSION+set}" != "set"; then
 		scm=$(scm_version --short)
-		res="$res${scm:++}"
+		res="$res"
 	fi
 fi
 
-- 
2.25.1

